src/app/src/main/java/com/example/bookmark/ExploreActivity.java

```diff
@@ -1,118 +1,103 @@
-                // Opens search fragment
-                new SearchDialogFragment().show(getSupportFragmentManager(),
-                    "SEARCH_AVAILABLE_BOOKS");
-            case R.id.menu_filter_search_filter_btn:
-                // TODO open filter fragment
-                break;
-        }
-        return (super.onOptionsItemSelected(item));
+    /**
+     * Returns whether the the owner field of each Book listed in the
+     * visibleBooksListView for this activity should be visible.
+     *
+     * @return boolean
+     */
+    @Override
+    protected boolean getBookOwnerVisibility() {
+        return true;
     }
 
+    /**
+     * Returns whether the the status field of each Book listed in the
+     * visibleBooksListView for this activity should be visible.
+     *
+     * @return boolean
+     */
     @Override
-    public void sendSearchedKeywords(String searchString) {
-        // Instead of sending intent to this same activity just calls
-        // executeSearch again
-
-        // TODO replace line below with searchString when we want to start
-        //  using search
-        executeSearch("");
+    protected boolean getBookStatusVisibility() {
+        return true;
     }
 
-    public void executeSearch(String searchedKeywords) {
+    /**
+     * Gets all books from the firestore database that are not accepted or
+     * borrowed and sets the values of visibleBooks and relevantBooks
+     * accordingly.
+     */
+    @Override
+    protected void getBooks() {
         StorageServiceProvider.getStorageService().retrieveBooks(books -> {
-            searchResults.clear();
+            visibleBooks.clear();
+            relevantBooks.clear();
             for (Book book : books) {
-                if (book.getDescription().contains(searchedKeywords)) {
-                    searchResults.add(book);
+                if ((book.getOwnerId() != user.getId()) &&
+                    (book.getStatus() != Book.Status.BORROWED) &&
+                    (book.getStatus() != Book.Status.ACCEPTED)) {
+                    relevantBooks.add(book);
                 }
             }
-            searchResultsAdapter.notifyDataSetChanged();
-        }, e -> DialogUtil.showErrorDialog(this, e));
+            visibleBooks.addAll(relevantBooks);
+            visibleBooksAdapter.notifyDataSetChanged();
+        }, e -> {
+            DialogUtil.showErrorDialog(this, e);
+        });
+    }
+
+    /**
+     * Returns the context that is used for the starting point of the
+     * intent that is created when a Book in the visibleBooksListView is
+     * clicked.
+     *
+     * @return Context
+     */
+    @Override
+    protected Context getPackageContext() {
+        return ExploreActivity.this;
+    }
+
+    /**
+     * Returns the class that is used for the destination of the
+     * intent that is created when a Book in the visibleBooksListView is
+     * clicked.
+     *
+     * @return Class<?></?>
```

Comment:
Not sure the closing tag </?> was meant to be inserted here. Probably happened due to autocomplete.

Reply:
Oh yea that was autocomplete lol Ill fix it